### PR TITLE
Stacking annotations

### PR DESCRIPTION
**Pull request by exys666**

### CODE REVIEW

File: src/main/java/com/futureprocessing/documentjuggler/annotation/AnnotationProcessor.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package com.futureprocessing.documentjuggler.annotation;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
+import java.util.HashSet;
+import java.util.Set;
+
+public final class AnnotationProcessor {
```
Comment (szymonsobocik) [inline src/main/java/com/futureprocessing/documentjuggler/annotation/AnnotationProcessor.java]: I would prefer it to be named `AnnotationReader` or something like this.
This class on reads annotations.

File: src/main/java/com/futureprocessing/documentjuggler/annotation/AnnotationProcessor.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    private final AnnotatedElement element;
+
+    public static AnnotationProcessor process(AnnotatedElement method) {
```
Comment (szymonsobocik) [inline src/main/java/com/futureprocessing/documentjuggler/annotation/AnnotationProcessor.java]: I would use different name: `read(...)`

File: src/main/java/com/futureprocessing/documentjuggler/annotation/AnnotationProcessor.java (hunk 1 above)
Comment (exys666) [inline src/main/java/com/futureprocessing/documentjuggler/annotation/AnnotationProcessor.java]: Maybe AnnotationFInder?

File: src/main/java/com/futureprocessing/documentjuggler/annotation/AnnotationProcessor.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        return new AnnotationProcessor(method);
+    }
+
+    private AnnotationProcessor(AnnotatedElement element) {
+        this.element = element;
+    }
+
+    private <A extends Annotation> A get(Class<A> annotationClass, AnnotatedElement element, Set<Annotation> checked) {
```
Comment (szymonsobocik) [inline src/main/java/com/futureprocessing/documentjuggler/annotation/AnnotationProcessor.java]: `checked` Set is always empty.

File: src/main/java/com/futureprocessing/documentjuggler/annotation/AnnotationProcessor.java (hunk 1 above)
Comment (szymonsobocik) [inline src/main/java/com/futureprocessing/documentjuggler/annotation/AnnotationProcessor.java]: Finder is also ok.

**Comment by szymonsobocik** [comment]
+1

**Pull request merged**